cloudferrylib/os/actions/networks_transporter.py

```diff
@@ -13,16 +13,73 @@
 # limitations under the License.
 
 
+from fabric.api import settings
+
+from cloudferrylib.base.action import action
 from cloudferrylib.base.action import transporter
-from cloudferrylib.utils import utils as utl
+from cloudferrylib.utils import utils
+from cloudferrylib.utils import remote_runner
+
+
+NAMESPACE = 'routers'
 
 
 class NetworkTransporter(transporter.Transporter):
 
     def run(self, **kwargs):
-        src_resource = self.src_cloud.resources[utl.NETWORK_RESOURCE]
-        dst_resource = self.dst_cloud.resources[utl.NETWORK_RESOURCE]
+        src_resource = self.src_cloud.resources[utils.NETWORK_RESOURCE]
+        dst_resource = self.dst_cloud.resources[utils.NETWORK_RESOURCE]
         info = src_resource.read_info()
         dst_resource.deploy(info)
         return {}
 
+
+class ChangeRouterBase(action.Action):
+    def __init__(self, *args, **kwargs):
+        super(ChangeRouterBase, self).__init__(*args, **kwargs)
+        self.net = self.cloud.resources[utils.NETWORK_RESOURCE]
+        self.cfg = self.cloud.cloud_config.cloud
+
+    def modify_router(self, router):
+        node = self.net.get_router_host(router['id'])
+        runner = remote_runner.RemoteRunner(node, self.cfg.ssh_user,
+                                            password=self.cfg.ssh_sudo_password)
+        with settings(abort_on_prompts=True,
+                      gateway=self.cloud.getIpSsh()):
+            id = router['id']
+            _, ext_cidr_len = router['ext_cidr'].split('/')
+            int_cidr = router['int_cidr']
+            ext_ip = router['ext_ip']
+            new_ip = router['new_ip']
+            device = runner.run("ip netns exec qrouter-%s ip addr | grep %s"
+                                " | awk '{ print $NF }'" %
+                                (router['id'], router['ext_ip']))
+            runner.run("ip netns exec qrouter-%s iptables -D "
+                       "quantum-l3-agent-snat -s %s -t nat -j SNAT "
+                       "--to-source %s" % (id, int_cidr, ext_ip))
+            runner.run("ip netns exec qrouter-%s ip addr del %s/%s dev %s" %
+                       (id, ext_ip, ext_cidr_len, device))
+            runner.run("ip netns exec qrouter-%s ip addr add %s/%s dev %s" %
+                       (id, new_ip, ext_cidr_len, device))
+            runner.run("ip netns exec qrouter-%s iptables -A "
+                       "quantum-l3-agent-snat -s %s -t nat -j SNAT "
+                       "--to-source %s" % (id, int_cidr, new_ip))
+
+
+class ChangeRouter(ChangeRouterBase):
```

Comment:
Please add docstring describing action: why it's needed, what problem it solves, what are the dependencies and how to restore original state if something goes wrong.